packages/time-picker/src/TimePicker/utils/generateRegexFromInput.ts

```diff
@@ -0,0 +1,24 @@
+import { DateSegment } from "@react-stately/datepicker"
+
+export const generateRegexFromInput = (segments: DateSegment[]): RegExp => {
+  const regexString = segments
+    .map(segment => {
+      if (segment.isPlaceholder || segment.type === "literal") {
+        if (segment.type === "hour") {
+          return ".{1,2}"
+        }
+        // replaces all placeholder text characters with regex match any characters
+        return Array.from(Array(segment.text.length).keys())
```

Comment:
Could this be achieve with a regex like `.{,${segment.text.length}}`

Or something a bit clearer?